command/bridge/deploy/deploy.go

```diff
@@ -604,73 +295,69 @@ func deployContracts(outputter command.OutputFormatter, client *jsonrpc.EthClien
 		})
 	}
 
-	if err := g.Wait(); err != nil {
-		return collectResultsOnError(results), err
-	}
-
-	commandResults := make([]command.CommandResult, 0, len(results))
-
-	for _, result := range results {
-		commandResults = append(commandResults, result)
-
-		populatorFn, exists := metadataPopulatorMap[result.Name]
-		if !exists {
-			continue
-		}
-
-		populatorFn(rootchainConfig, result.Address)
-	}
-
-	g, ctx = errgroup.WithContext(cmdCtx)
-
-	for contractName := range results {
-		contractName := contractName
-
-		initializer, exists := initializersMap[contractName]
-		if !exists {
-			continue
+	initializeContractFn := func(contract *contract, txr txrelayer.TxRelayer, chainID int64) {
+		if contract.initializeFn == nil {
+			// some contracts do not have initialize function
+			return
 		}
 
 		g.Go(func() error {
 			select {
-			case <-cmdCtx.Done():
-				return cmdCtx.Err()
+			case <-ctx.Done():
+				return ctx.Err()
 			default:
-				return initializer(outputter, txRelayer, initialValidators, rootchainConfig, deployerKey, chainID)
+				return contract.initializeFn(outputter, externalTxRelayer,
+					initialValidators, bridgeConfig, deployerKey, chainID)
+
 			}
 		})
 	}
 
+	// deploy external contracts
+	for _, contract := range externalContracts {
+		deployContractFn(contract, externalTxRelayer)
+	}
+
+	// deploy internal contracts if blade is live
+	if !params.isBootstrap {
+		for _, contract := range internalContracts {
+			deployContractFn(contract, internalTxRelayer)
+		}
+	}
```

Comment:
```suggestion

	for _, contract := range internalContracts {
		deployContractFn(contract, internalTxRelayer)
	}

```

Reply:
We are only deploying internal contracts if blade is live (`isBootstrap` is false), if it is not live, we do the pre-allocate.